ui/pages/DataManagement/reducers.js

```diff
@@ -64,15 +74,19 @@ const loadMultipleData = (path, getUpdateData, dispatchType, formatSuccessMessag
   ).post(values).then(() => {
     const { info, warnings } = successResponseJson
     let numLoaded = 0
-    return Promise.all(getUpdateData(successResponseJson, values).map(
-      ([entityUrl, entityId, body]) => new HttpRequestHelper(
-        entityUrl,
-        () => {
-          numLoaded += 1
-        },
-        e => warnings.push(`Error loading ${entityId}: ${e.body && e.body.error ? e.body.error : e.message}`),
-      ).post(body),
-    )).then(() => {
+    const updateData = getUpdateData(successResponseJson, values)
+    return updateData.reduce((prevPromise, item, index) => {
```

Comment:
it might be cleaner to reason about this by splitting the `batch`-ing from the `Promise` stuff with a `batch(arr, size)` helper.  that's usually how I've seen this... 🤷 but this definitely looks right as is!

Reply:
I actually had it like that originally but it was basically a `reduce` followed by another nearly identical `reduce` so it looked messier to me than having the single function call

Reply:
yep that's fair!  I know it doesn't fit the style of the reducers but something like this w/ a generator might also work:

```
function* batch(arr, size){
  for(let i = 0; i < arr; i += n) yield arr(i, i + size);
}
```